numba/parfors/parfor_lowering.py

```diff
@@ -1634,28 +1682,23 @@ def load_range(v):
         builder.store(stop, builder.gep(dim_stops,
                                         [context.get_constant(types.uintp, i)]))
 
-    sched_size = get_thread_count() * num_dim * 2
-    sched = cgutils.alloca_once(
-        builder, sched_type, size=context.get_constant(
-            types.uintp, sched_size), name="sched")
-    debug_flag = 1 if config.DEBUG_ARRAY_OPT else 0
-    scheduling_fnty = llvmlite.ir.FunctionType(
-        intp_ptr_t, [uintp_t, sched_ptr_type, sched_ptr_type, uintp_t, sched_ptr_type, intp_t])
-    if index_var_typ.signed:
-        do_scheduling = cgutils.get_or_insert_function(builder.module,
-                                                       scheduling_fnty,
-                                                       "do_scheduling_signed")
-    else:
-        do_scheduling = cgutils.get_or_insert_function(builder.module,
-                                                       scheduling_fnty,
-                                                       "do_scheduling_unsigned")
+    get_chunksize = cgutils.get_or_insert_function(
+        builder.module,
+        llvmlite.ir.FunctionType(uintp_t, []),
+        name="get_parallel_chunksize")
+
+    set_chunksize = cgutils.get_or_insert_function(
+        builder.module,
+        llvmlite.ir.FunctionType(llvmlite.ir.VoidType(), [uintp_t]),
+        name="set_parallel_chunksize")
 
     get_num_threads = cgutils.get_or_insert_function(
         builder.module,
         llvmlite.ir.FunctionType(llvmlite.ir.IntType(types.intp.bitwidth), []),
         "get_num_threads")
 
     num_threads = builder.call(get_num_threads, [])
+    current_chunksize = builder.call(get_chunksize, [])
```

Comment:
I think this new IR generation would benefit from some commentary regarding the handling of the chunksize. It looks like the current chunksize is stored, then later, once the divisions have been calculated, the chunksize is zeroed during  kernel execution and then finally restored to the original value post execution of the kernel. What's the purpose of doing this? Is it to ensure subsequent chunksize use is zero?

Reply:
I think I am forgetting some details here but what I remember is that the idea is to create an invariant that 0 is the default (which the docs state).  I don't remember if there was some issue such that nested cases had a problem if you don't do this reset to 0 at this boundary or if I made the decision that the invariant should also be true inside parfor regions because it is unlikely that a nested parfor would need the same chunksize.  I'm open to discussion here.  Please advise. @stuartarchibald